selftests/functional/plugin/spawners/test_podman.py

```diff
@@ -0,0 +1,43 @@
+import os
+import shutil
+import unittest
+
+from avocado.utils import process, script
+from selftests.utils import AVOCADO, TestCaseTmpDir
+
+TEST_INSTRUMENTED_PASS = """from avocado import Test
+
+
+class PassTest(Test):
+
+    def test(self):
+        pass
+"""
+
+
+@unittest.skipIf(shutil.which('podman') is None,
+                 "Podman not installed (command podman is missing)")
+class PodmanSpawnerTest(TestCaseTmpDir):
+
+    def test_avocado_instrumented(self):
+        test = script.Script(os.path.join(self.tmpdir.name, "passtest.py"),
```

Comment:
I'd use a context manager here to avoid the `test.save()`:

```diff
diff --git a/selftests/functional/plugin/spawners/test_podman.py b/selftests/functional/plugin/spawners/test_podman.py
index 4209085fc..e303d15f4 100644
--- a/selftests/functional/plugin/spawners/test_podman.py
+++ b/selftests/functional/plugin/spawners/test_podman.py
@@ -20,14 +20,13 @@ class PassTest(Test):
 class PodmanSpawnerTest(TestCaseTmpDir):
 
     def test_avocado_instrumented(self):
-        test = script.Script(os.path.join(self.tmpdir.name, "passtest.py"),
-                             TEST_INSTRUMENTED_PASS)
-        test.save()
-        result = process.run(f"{AVOCADO} run "
-                             f"--job-results-dir {self.tmpdir.name} "
-                             f"--disable-sysinfo --nrunner-spawner=podman "
-                             f"--spawner-podman-image=fedora:latest -- "
-                             f"{test}")
+        with script.Script(os.path.join(self.tmpdir.name, "passtest.py"),
+                           TEST_INSTRUMENTED_PASS) as test:
+            result = process.run(f"{AVOCADO} run "
+                                 f"--job-results-dir {self.tmpdir.name} "
+                                 f"--disable-sysinfo --nrunner-spawner=podman "
+                                 f"--spawner-podman-image=fedora:latest -- "
+                                 f"{test}")
         self.assertTrue(result.exit_status == 0)
         self.assertIn("passtest.py:PassTest.test: STARTED", result.stdout_text)
         self.assertIn("passtest.py:PassTest.test:  PASS", result.stdout_text)
```

Or simply reuse `examples/tests/passtest.py`:

```diff
diff --git a/selftests/functional/plugin/spawners/test_podman.py b/selftests/functional/plugin/spawners/test_podman.py
index 4209085fc..2181d651e 100644
--- a/selftests/functional/plugin/spawners/test_podman.py
+++ b/selftests/functional/plugin/spawners/test_podman.py
@@ -2,32 +2,20 @@ import os
 import shutil
 import unittest
 
-from avocado.utils import process, script
+from avocado.utils import process
 from selftests.utils import AVOCADO, TestCaseTmpDir
 
-TEST_INSTRUMENTED_PASS = """from avocado import Test
-
-
-class PassTest(Test):
-
-    def test(self):
-        pass
-"""
-
 
 @unittest.skipIf(shutil.which('podman') is None,
                  "Podman not installed (command podman is missing)")
 class PodmanSpawnerTest(TestCaseTmpDir):
 
     def test_avocado_instrumented(self):
-        test = script.Script(os.path.join(self.tmpdir.name, "passtest.py"),
-                             TEST_INSTRUMENTED_PASS)
-        test.save()
         result = process.run(f"{AVOCADO} run "
                              f"--job-results-dir {self.tmpdir.name} "
                              f"--disable-sysinfo --nrunner-spawner=podman "
                              f"--spawner-podman-image=fedora:latest -- "
-                             f"{test}")
+                             f"examples/tests/passtest.py")
         self.assertTrue(result.exit_status == 0)
         self.assertIn("passtest.py:PassTest.test: STARTED", result.stdout_text)
         self.assertIn("passtest.py:PassTest.test:  PASS", result.stdout_text)
```